src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp

```diff
@@ -45,109 +45,317 @@
+        int AirflowNetworkNumOfSurCracks = 0;
+        int AirflowNetworkNumOfSurELA = 0;
+        int AirflowNetworkNumOfSFR = 0;
+        int AirflowNetworkNumOfExtNode = 0;
+        int AirflowNetworkNumOfOutAirNode = 0;
+        int AirflowNetworkNumOfSingleSideZones = 0; // Total number of zones with advanced single sided wind pressure coefficient calculation
+        int DisSysNumOfNodes = 0;
+        int DisSysNumOfLeaks = 0;
+        int DisSysNumOfELRs = 0;
+        int DisSysNumOfDucts = 0;
+        int DisSysNumOfDuctViewFactors = 0;
+        int DisSysNumOfDampers = 0;
+        int DisSysNumOfCVFs = 0;
+        int DisSysNumOfDetFans = 0;
+        int DisSysNumOfCoils = 0;
+        int DisSysNumOfHXs = 0;
+        int DisSysNumOfCPDs = 0;
+        int DisSysNumOfTermUnits = 0;
+        int DisSysNumOfLinks = 0;
+        int NumOfExtNodes = 0;
+        Real64 IncAng = 0.0;                     // Wind incidence angle relative to facade normal (deg)
+        int SupplyFanType = 0;                   // Supply air fan type
+        Real64 MaxOnOffFanRunTimeFraction = 0.0; // max Run time fraction for an On/Off fan flow rate among airloops
+        Real64 CurrentEndTimeLast = 0.0;         // last end time
+        Real64 TimeStepSysLast = 0.0;            // last system time step
+        int AirflowNetworkNumOfOccuVentCtrls = 0;
+        int IntraZoneNumOfNodes = 0;
+        int IntraZoneNumOfLinks = 0;
+        int IntraZoneNumOfZones = 0;
+        int NumOfPressureControllers = 0; // number of pressure controllers
+        int NumOfOAFans = 0;              // number of OutdoorAir fans
+        int NumOfReliefFans = 0;          // number of OutdoorAir relief fans
+        bool AirflowNetworkGetInputFlag = true;
+        bool AssignFanAirLoopNumFlag = true;
+        bool ValidateDistributionSystemFlag = true;
+        Array1D<Real64> FacadeAng =
+            Array1D<Real64>(5); // Facade azimuth angle (for walls, angle of outward normal to facade measured clockwise from North) (deg)
+        Array1D<Real64> LoopPartLoadRatio;
+        Array1D<Real64> LoopOnOffFanRunTimeFraction;
+        Array1D<bool> LoopOnOffFlag;
+
+        bool ValidateExhaustFanInputOneTimeFlag = true;
+        bool initializeOneTimeFlag = true;
+        bool initializeMyEnvrnFlag = true;
+        bool CalcAirflowNetworkAirBalanceOneTimeFlag = true;
+        bool CalcAirflowNetworkAirBalanceErrorsFound = false;
+        bool UpdateAirflowNetworkMyOneTimeFlag = true;
+        bool UpdateAirflowNetworkMyOneTimeFlag1 = true;
+
+        // CalcAirflowNetworkAirBalance variables
+        int ErrCountVar = 0;
+        int ErrCountHighPre = 0;
+        int ErrCountLowPre = 0;
+        int ErrIndexHighPre = 0;
+        int ErrIndexVar = 0;
+        int ErrIndexLowPre = 0;
+
+        // Object Data
+        EPVector<AirflowNetwork::AirflowNetworkReportVars> AirflowNetworkZnRpt;
+        std::unordered_map<std::string, std::string> UniqueAirflowNetworkSurfaceName;
```

Comment:
Not a huge fan of using `map` in general, but could be a legitimate use. How many surfaces are typically in this structure? All surfaces in the model, i.e., hundreds or thousands? Only surfaces with openings? Also, if you are not mapping to anything, you can use `unordered_set`.

Reply:
I think this one gets used in `VerifyUniqueInterObjectName`? I believe that is used to make sure that no two AFN surfaces refer to the same heat transfer surface. That might be something that should be done in schema validation?

Reply:
Oh, and the number of elements in there is going to be on the order of the number of surfaces in the model.

Reply:
I don't know JSON schema very well, but I think this would be hard to do in validation. It's not a unique key situation, it's a uniqueness constraint on an arbitrary field. How would you even specify the level of the schema at which the field would have to be unique?

Reply:
Over in XSD land, one can definitely apply those sorts of constraints on fields. I don't know if you can require a field to be unique and also be from another set of elements. Maybe that's not possible with JSON schema? If it is it is probably complicated. It's pretty complicated with XSD, IIRC. In any case, I'm not sure why this needs a map when a vector of all the used names should be sufficient, but maybe I'm missing something.

Reply:
Vector is fine too. Map is only better when you have many elements to search through. Also, these kinds of searches should generally be done with surface indices rather than surface names, and for integers `std::vector` is 100% better even for many elements (or at least as many as you are likely to get in EnergyPlus).